src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -297,33 +295,216 @@ export class ResultsViewPageStore {
-        await: () => [
-            this.genes,
-            this.selectedMolecularProfiles,
-            this.mutationMapperStores,
-            this.molecularData,
-            this.defaultOQLQuery,
-            this.geneToMutData
+    readonly unfilteredAlterations = remoteData<AlterationData[]>({
+        await: ()=>[
+            this.mutations,
+            this.molecularData
         ],
-        invoke: async() => {
+        invoke: ()=>{
+            return Promise.resolve(([] as AlterationData[]).concat(this.mutations.result!).concat(this.molecularData.result!));
+        }
+    });
 
-            const filteredMolecularDataByGene = _.groupBy(this.molecularData.result, (item: GeneMolecularData) => item.gene.hugoGeneSymbol);
+    readonly filteredAlterations = remoteData<AlterationData[]>({
+        await:()=>[
+            this.unfilteredAlterations,
+            this.selectedMolecularProfiles,
+            this.defaultOQLQuery
+        ],
+        invoke:()=>{
+            return Promise.resolve(
+                filterCBioPortalWebServiceData(this.oqlQuery, this.unfilteredAlterations.result!, (new accessors(this.selectedMolecularProfiles.result!)), this.defaultOQLQuery.result!, true)
+            );
+        }
+    });
 
-            const genesAsDictionary = _.keyBy(this.genes.result, (gene: Gene) => gene.hugoGeneSymbol);
+    readonly filteredAlterationsByOQLLine = remoteData<OQLFilteredLine<AlterationData>[]>({
+        await: ()=>[
+            this.unfilteredAlterations,
+            this.selectedMolecularProfiles,
+            this.defaultOQLQuery
+        ],
+        invoke: ()=>{
+            return Promise.resolve(filterCBioPortalWebServiceDataByOQLLine(this.oqlQuery, this.unfilteredAlterations.result!,
+                (new accessors(this.selectedMolecularProfiles.result!)), this.defaultOQLQuery.result!, true));
+        }
+    });
 
-            //TODO must check state
-            // now merge alterations with mutations by gene
-            const mergedAlterationsByGene = _.mapValues(genesAsDictionary, (gene: Gene) => {
-                // if for some reason it doesn't exist, assign empty array;
-                return _.concat(([] as (Mutation | GeneMolecularData)[]), this.geneToMutData.result![gene.hugoGeneSymbol],
-                    filteredMolecularDataByGene![gene.hugoGeneSymbol!]!);
-            });
-            const ret = _.mapValues(mergedAlterationsByGene, (mutations: (Mutation | GeneMolecularData)[]) => {
-                return filterCBioPortalWebServiceData(this.oqlQuery, mutations, (new accessors(this.selectedMolecularProfiles.result!)), this.defaultOQLQuery.result!, undefined, true);
+    readonly caseAggregatedData = remoteData<CaseAggregatedData<AlterationData>>({
+        await: ()=>[
+            this.filteredAlterations,
+            this.samples,
+            this.patients
+        ],
+        invoke: ()=>{
+            return Promise.resolve({
+                samples:
```

Comment:
something kind weird about formatting here

Reply:
its for line length